framework/src/materials/MaterialPropertyInterface.C

```diff
@@ -182,3 +182,10 @@ MaterialPropertyInterface::statefulPropertiesAllowed(bool stateful_allowed)
 {
   _stateful_allowed = stateful_allowed;
 }
+
+void
+MaterialPropertyInterface::checkExecutionStage()
+{
+  if (_mi_feproblem.startedInitialSetup())
+    mooseWarning("Material properties should be gotten at construction time.");
```

Comment:
Overall I like these changes... but can we change the message so that it doesn't use "gotten"?  How about:

`Material properties should be retrieved during object construction.`

:smile: